system-reinstall-bootc/src/prompt.rs

```diff
@@ -92,6 +93,42 @@ pub(crate) fn ask_yes_no(prompt: &str, default: bool) -> Result<bool> {
         .context("prompting")
 }
 
+pub(crate) fn mount_warning() -> Result<()> {
+    let mounts = bootc_mount::run_findmnt(&[], None)?;
+
+    let problematic_filesystems: Vec<&Filesystem> = mounts
+        .filesystems
+        .iter()
+        .find(|mount| mount.target == "/")
+        .map(|root| {
+            root.children
+                .iter()
+                .flatten()
+                .filter(|child| child.source == root.source)
+                .collect()
+        })
+        .unwrap_or_default();
+
+    if !problematic_filesystems.is_empty() {
+        println!();
+        println!("Notice: the following filesystems are currently mounted on the root disk. After rebooting into the bootc system, these filesystems will not be mounted. They will be preserved so you will need to manually mount or remove them.");
```

Comment:
Well...strictly speaking, someone could inject configuration into their image to mount it. It would be totally sane to me for example to have an OS configuration that auto-discovers a btrfs subvolume named `home` and mounts it at `/home` or whatever.

Getting into this level of detail, we could in theory also scrape `/etc/fstab` and the `.mount` units in the target environment...but, yeah a lot there.

I'm uncertain. I think in the short term given the above, effectively special casing "/home btrfs subvolume" could make sense as a starting point.

> NOTE: Detected mounted btrfs subvolumes: /home
>   These subvolumes are left unchanged by this tool.

?